### PR TITLE
Added HW 4

### PR DESCRIPTION
**Pull request by asahirov**

### CODE REVIEW

File: tests/asahirov/test_homework_4.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    data = [{"age": 16, "name": "John", "sex": "M"},
+            {"age": 34, "name": "Marry", "sex": "F"},
+            {"age": 25, "name": "Mathew", "sex": "M"}]
+
+    sorted_list = sorted(data, key=lambda i: i["age"])
+    assert sorted_list[-1]["age"] == 34
+
+    sorted_list_reverse = sorted(data, key=lambda i: i["age"], reverse=True)
+    assert sorted_list_reverse[-1]["age"] == 16
+
```
Comment (fkudinov-mn) [inline tests/asahirov/test_homework_4.py:15]: assert [i["age"] for i in sorted_list] == [34, 25, 16]

File: tests/asahirov/test_homework_4.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    sorted_list = sorted(data, key=lambda i: (i["sex"] == "M", i["age"]))
+    assert sorted_list[0]["name"] == "Marry"
+    assert sorted_list[-1]["name"] == "Mathew"
+
+
+def test_comprehensions_dict():
+    some_dict = {"a": 1, "b": 2, "c": 3, "d": 4, "e": 5}
+
+    updated_dict = [i[0] for i in some_dict.items() if i[1] % 2 != 0]
```
Comment (fkudinov-mn) [inline tests/asahirov/test_homework_4.py:30]: also remember we do not have strong guaranttes about  keys order, so better to sort before assertion


    updated_dict = [k for k, v in some_dict.items() if v % 2 != 0]

    updated_dict = [k for (k, v) in some_dict.items() if v % 2 != 0]

**Review by fkudinov-mn — APPROVED** [review]

File: tests/asahirov/test_homework_4.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    # sort data (same for min max)
+    data = [{"age": 16, "name": "John", "sex": "M"},
+            {"age": 34, "name": "Marry", "sex": "F"},
+            {"age": 25, "name": "Mathew", "sex": "M"}]
+
+    sorted_list = sorted(data, key=lambda i: i["age"])
+    assert sorted_list[-1]["age"] == 34
+
+    sorted_list_reverse = sorted(data, key=lambda i: i["age"], reverse=True)
+    assert sorted_list_reverse[-1]["age"] == 16
```
Comment (fkudinov-mn) [inline tests/asahirov/test_homework_4.py:14]: add min max tests also

File: tests/asahirov/test_homework_4.py (hunk 1 above)
Comment (asahirov) [reply tests/asahirov/test_homework_4.py:15]: updated

File: tests/asahirov/test_homework_4.py (hunk 2 above)
Comment (asahirov) [reply tests/asahirov/test_homework_4.py]: updated

File: tests/asahirov/test_homework_4.py (hunk 3 above)
Comment (asahirov) [reply tests/asahirov/test_homework_4.py]: updated

**Review by fkudinov-mn — APPROVED** [review]

**Pull request merged**